llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -939,6 +946,7 @@ bool VPInstruction::onlyFirstLaneUsed(const VPValue *Op) const {
   case Instruction::ICmp:
   case Instruction::Select:
   case Instruction::Or:
+  case Instruction::Freeze:
```

Comment:
We should probably also add this opcode to `VPInstruction::opcodeMayReadOrWriteFromMemory` and return `false`.